### PR TITLE
Geomap: use name as UID

### LINKED ISSUE #41504: Geomap: Remove layer UID in favor of using name as unique identifier
Verify unique layer names and remove UID from layer element

As part of this clean up layer actions model without use of uid (see changes in https://github.com/grafana/grafana/pull/41491 related to actions)

### PR DESCRIPTION
**Pull request by ryantxu**
Fixes #41504 and #41511

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/panel/geomap/GeomapPanel.tsx [hunk 1]
```diff
@@ -195,6 +196,11 @@ export class GeomapPanel extends Component<Props, State> {
       this.layers = result;
 
       this.doOptionsUpdate(endIndex);
+
+      // Add the layers in the right order
+      const group = this.map?.getLayers()!;
+      group.clear();
+      this.layers.forEach((v) => group.push(v.layer));
```
Comment (ryantxu) [inline public/app/plugins/panel/geomap/GeomapPanel.tsx:203]: ^^ fixes #41511

File: public/app/plugins/panel/geomap/editor/LayersEditor/LayerHeader.tsx [hunk 2]
```diff
@@ -1,17 +1,15 @@
...
-import { GrafanaTheme } from '@grafana/data';
-
-import { MapLayerState } from '../../types';
+import { GrafanaTheme, MapLayerOptions } from '@grafana/data';
 
 export interface LayerHeaderProps {
-  layer: MapLayerState<any>;
-  layers: Array<MapLayerState<any>>;
-  onChange: (layer: MapLayerState<any>) => void;
+  layer: MapLayerOptions<any>;
```
Comment (ryantxu) [inline public/app/plugins/panel/geomap/editor/LayersEditor/LayerHeader.tsx:7]: changes from the internal `state` object, to just editing the JSON model

File: public/app/plugins/panel/geomap/GeomapPanel.tsx [hunk 3]
```diff
@@ -354,28 +361,46 @@ export class GeomapPanel extends Component<Props, State> {
...
+
+      uid = newOptions.name;
+      this.byName.set(uid, current);
+    }
+
+    // Type changed -- requires full re-initalization
+    if (current.options.type !== newOptions.type) {
+      // full init
+    } else {
+      // just update options
```
Comment (ryantxu) [inline public/app/plugins/panel/geomap/GeomapPanel.tsx:396]: TODO -- (but not this PR!)

We currently do full initialization on any change.  We *should* only need to do that when the type changes

**Review by nikki-kiga — APPROVED** [review]
LGTM

**Pull request merged**